### DISCUSSION TITLE
Remove podcast:images

### DISCUSSION THREAD

**Discussion by jamescridland** [Podcastindex-org/podcast-namespace#584] [Deprecation Proposal] [answered]
The new podcast namespace [defines `podcast:images`](https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md#images), a way to define additional sized images. There are a few drawbacks with this specification:

1. It isn't used by podcast apps. To my knowledge, literally one podcast app is using the spec, PodLP.

2. It isn't used by podcasters. There are [49 feeds](https://podcastindex.social/@js/111302809859677570) with this tag at the episode/item level. That represents 0.001149% of all podcasts.

3. It is limited only to square thumbnails. Alternative images could be useful for podcast apps and directories: @redimongo wanted a 16:9 banner here: #527 

4. It is a duplication [of an existing specification](https://www.rssboard.org/media-rss), `media:thumbnails`, which is used by YouTube. Podcasters can define a 1280x720 image using `media:thumbnails` which gets used by YouTube's RSS ingestion tool. That is part of the media RSS specification, which is part of the RSS advisory board. Already, over 9000 existing feeds have a media:thumbnail tag at the episode/item level. That's 183 times more than are using the podcast:images tag.

**I would like to propose that `podcast:images` is removed.**

It won't make any impact on any podcast app, since podcast apps will fall back to the standard image tags. It won't make any impact on podcasters, almost none of whom are using this tag. Its "replacement" is more capable, and, importantly, also in use.

As a matter of policy, I believe that we should support tags that are used by the community - but remove tags that go un-used, or that we have made a mistake with. That would be the sign of a mature specification and ecosystem. Since we have empirical evidence that this tag is unused after more than 12 months in the specification, it's hard to argue that it should remain in a specification.

**Comment by nathangathright** [comment]
Can you expand on how removing an unused tag from the namespace is a benefit for the community? I’m having trouble understanding how keeping it takes away support from any other tags. On the other hand, I have concerns that removing tags will set a precedent where developers will take a "wait and see" approach to make sure their efforts won’t be wasted.

**Comment by jamescridland** [comment]
We can, of course, keep a tag which is unused. No harm comes from filling
the podcast namespace with all kinds of tags that nobody uses, and that are
duplicates of objectively better tags which are. Where storage is free, we
can come up with all kinds of ideas, and even if they fail to be adopted,
keep them as part of the specification.

But, I believe we're better than that. The namespace - **our** namespace -
should be curated to *only* offer features that advance podcasting. We
should be proud enough to allow input from elsewhere, and also proud enough
to keep our specification compact and focused.

By making it easy to support *all* features in Podcasting 2.0 - and not
have to sift through a raft of unused, badly-drafted features to work out
which are actually in-use - we will encourage adoption of the namespace as
a whole.

> I have concerns that removing tags will set a precedent where developers
will take a "wait and see" approach to make sure their efforts won’t be
wasted.

Of course, this is the approach many developers have taken anyway. You
could argue that Libsyn, Megaphone and ART19 have taken that approach to
the entire project. And, who could blame them - tags like this, which are
unused, are a reason why they would think twice about any form of support.

I believe that every feature on Podcasting 2.0 should have an "owner", who
evangelises about the feature and works to get it implemented. If we fail
to get a feature enthusiastically adopted by the community, we should
examine the reasons why, and retire that feature.

The reasons why here are:
1. It was an overly open specification, with no recommendations for use, or
examples of use-cases. It's unclear why you'd ever want to use it. We
should learn from that.
2. It was also a specification which was too constricting, since it didn't
take into account those that wanted images of different aspect ratios, or
even other formats like video. We should learn from this, too.
3. A feature existed already in the media namespace. It was well specified,
and could have been extended to specify filetypes. But we failed to do our
due diligence in researching it, and produced a less-capable duplicate.
4. It had no evangelist to promote its use; and as a consequence is only
implemented in one podcast app, and not by any podcast hosts. If it's not
supported by the Podcasting 2.0 podcast, that's a pretty large red flag
that the community doesn't want it.

I'm keen to see this namespace as a living, breathing thing: not a dumping
ground of unwanted failures. Deprecation of this feature would be the right
thing to do.

**Reply by redimongo** [reply]
@jamescridland please know I agree 100% with what you are saying. However can we also just point out that many tags being added are not for the benefit of podcasters but for music which goes against your "The namespace - **our** namespace - should be curated to *only* offer features that advance podcasting"

I personally questioned why PodcastIndex or this namespace would be promoting things like livestreams and music when that too is technically not the definition of Podcasts (in the traditional sense). 

I think the namespace will become bloated with useless tags and while am not against adding tags like livestream or others that grows the media landscape. 

I think the namespace might be getting lost with its mission of supporting podcasters. 

It's almost like another namespace should enter of music or even live-streaming. 

But then you're starting from scratch and gee who wants to do that when you can just add anything to this one. ☺️

Thoughts?

**Comment by tonytamsf** [comment]
I ran some numbers across all 4 million podcasts and found 647 podcasts with the `podcast:images` tag.  Most of them have it at the item level.  Given this is 10X more than the 49 claimed, does this change the conversation? Here is the list of podcasts https://github.com/tonytamsf/podcastindex_data/blob/main/data/images.podcasts.txt

**Reply by redimongo** [reply]
647 is still a really low number in comparison to over 4million. 

And I have to agree that the image tag is kinda obsolete when you think YouTube doesn't accept it but does accept the RSS media tag which is what images is meant to do.

**Reply by picard102** [reply]
what's the number using v4v?

**Comment by daveajones** [comment]
I'm going to ask questions in a methodical way that ensures I fully understand.  If the questions don't make sense, humour me please.  I wouldn't ask it if it wasn't necessary for me to get a full picture in my head.

First question:  Are you proposing that we remove any tag that doesn't reach a particular threshold within the "12 months" time frame you mentioned?

**Reply by jamescridland** [reply]
> Are you proposing that we remove any tag that doesn't reach a particular threshold within the "12 months" time frame you mentioned?

I'm proposing that we review, after an appropriate time, whether a particular feature has gained any traction - and, if it has not, to deprecate it.

"Deprecation" means removing from the main documents, shifting the documentation to a "deprecated" tag. In a validator, you might see a warning that the feature is deprecated. It would not be an error and would not fail validation - it's deprecated.

I'm not wedded to a 12 month timeframe. In fact, [the proposal](https://github.com/Podcastindex-org/podcast-namespace/issues/43) was made in October 2020. It's had more than **three years** for people to use it.

**Reply by daveajones** [reply]
How do we determine an appropriate time:adoption ratio?  We currently have no formal deprecation policy, so everything would have to be spelled out in black and white before taking action.

**Reply by jamescridland** [reply]
Taking one step back: should we adopt a policy of deprecating features that haven't gained adoption? I think we should.

But I suspect it might be sensible taking these on a case by case basis, and just as we accept proposals for new features, we also accept proposals for deprecated ones. I don't think a set timeframe is appropriate really.

**Reply by daveajones** [reply]
By it's nature, it wouldn't be something that's a rush.  So, as a start I can create a Discussion category for proposing deprecations and move the discussions there.

**Comment by jamescridland** [comment]
@daveajones I'd like to recommend that removing this tag is announced in phase 7.

It fails the test of being implemented in enough apps (as I understand it, it's only implemented in one); and it fails the test of being adopted by many podcasts after two years. It's a duplicate of an existing tag that existed before we worked on it, and that does a better job (since it deals with different aspect ratios).

**Comment by theDanielJLewis** [comment]
It's @jamescridland's #3 issue, "It is limited only to square thumbnails," that I think makes this feature almost useless.

I think we _do_ need a way to specify additional images, especially 16:9 and circle-friendly. That's what I _thought_ we could do with `<podcast:images>`, but apparently not.

Something to keep in mind regarding development is that it always goes two ways: the publishing tools must implement support for users (and highlight the new feature with education), and podcasters must use it. It's that second one that might be the bigger barrier because not all podcasters have alternate branding images, let alone even know there could be be a need for different images.

In this case, I care less about the _method_ for supporting alternative images and more about simply having the support that's flexible for current and future applications.

**Reply by jamescridland** [reply]
> I think we do need a way to specify additional images, especially 16:9 and circle-friendly. 

I agree, we do. We have it, [with the `media:thumbnails` tag](https://www.rssboard.org/media-rss#media-thumbnails) from the Media RSS namespace. Reinventing wheels aren't necessary here.

**Reply by theDanielJLewis** [reply]
I like that such an option already exists, but "reinventing wheels aren't necessary here" shouldn't apply in this case. `<media:thumbnail>` is part of a different namespace, and one of the core ideas of the Podcast Namespace is to make a single namespace with all the podcasting features. Thus, we _are_ reinventing several wheels already when compared to the "iTunes" namespace.

If the thumbnails tag was part of the core RSS spec, then I would agree we should use it instead of making an additional tag. But since it's a third-party tag, I think we should consider making a Podcast Namespace version.

_Unless_ the Media namespace is as ubiquitous as the Content namespace. But even then, Apple has already deprecated their support for the Content namespace, and it seems a bit pointless to promote another namespace solely for the use of one barely supported tag.

**Reply by ryan-lp** [reply]
> one of the core ideas of the Podcast Namespace is to make a single namespace with all the podcasting features.

Is it? That should be open for discussion to see if there is consensus on this. E.g. Would we also want to reinvent all of the itunes tags? I personally don't think we should be reinventing tags from other namespaces. Whoever invents a standard gets the naming rights for its namespace, and XML is designed to cooperatively permit multiple namespaces to be referenced within a single document. If you want to start copying other people's standards and rebranding it with your own namespace, I would like to see a separate proposal for that and a separate discussion around it. The Podcast Index / Podcasting 2.0 is not the only standards group in town, and it would be extremely uncooperative to act as if it is the only one by rebranding everyone else's standards as their own. Imagine if everyone else did that in reverse, pretending to be the only standard in town and copying everyone else's ideas and rebranding them within their own namespace. Namespaces are designed to allow different standards from different organisations to cooperate and coexist, and this would fly in the face of that. But I'll leave these arguments for a separate discussion if someone wants to make an official proposal. Of course there are going to be arguments on both sides, but that discussion needs to happen before it is claimed as a goal we all agree to.

**Reply by jamescridland** [reply]
> one of the core ideas of the Podcast Namespace is to make a single namespace with all the podcasting features

I'll be frank: I don't agree with this. The point of RSS is that you can use more than one namespace. There is no benefit to be gained by insisting that all other namespaces are "not invented here" and thus not useful.

The Media RSS namespace is being used by 9,000 podcasts. Just as we'd not recommend that the `itunes` namespace is no longer used, the right thing for podcasting is to continue using a specification that is in use and serves its purpose. We're not here to build a Podcast Namespace, we're here to benefit podcasting. Aren't we?

**Reply by theDanielJLewis** [reply]
It's all right there on this very repo's landing page.

> A **wholistic** rss namespace for podcasting that is meant to **synthesize the fragmented world of podcast namespaces**. The broad goal is to create a **single, compact, efficient namespace** that is easily extensible, community controlled/authored and addresses the needs of the independent podcast industry now and in the future. Our hope is that this namespace will become the framework that the independent podcast community needs to deliver new functionality to apps and aggregators. [[source](https://github.com/podcastindex-org/podcast-namespace#the-podcast-namespace), emphasis added]

> Goal #1 - **Eliminate Redundancy**
> There is significant overlap amongst the many existing podcast namespaces. Each platform and publisher has created their own namespace to give their respective system and audience the metadata they need in the way they want it delivered. [[source](https://github.com/podcastindex-org/podcast-namespace#goal-1---eliminate-redundancy), emphasis added]

Plus, it's been said many times by both @daveajones and Adam that they want the Podcast Namespace to eventually be a drop-in replacement so developers could get all the features they need from the Podcast Namespace instead of having to also use the "iTunes" namespace or any others.

**Reply by jamescridland** [reply]
Ah, well. *shrug*

Good luck with it all

**Reply by redimongo** [reply]
@jamescridland i take it you have a different view of the podcast namespace. 

I do agree we should not be reinventing the wheel.

(For others in this convo) 

As we know the likes of Spotify and Apple won't follow unless everyone tomorrow removed iTunes namespace from their podcast which I can tell you will not happen. 

We need to coexist.

**Reply by ryan-lp** [reply]
It looks like we've reached consensus in Australia, for starters :-)

I see no reason why the project's goals couldn't be changed to reflect the interests of its constituency. That sort of already happened with the mission statement which was changed from being about protecting free speech to being about open standards. We don't need to make this standard into a monopoly to serve that mission, it can coexist.

**Comment by Marzal** [comment, marked as answer]
> "Deprecation" means removing from the main documents, shifting the documentation to a "deprecated" tag. In a validator, you might see a warning that the feature is deprecated. It would not be an error and would not fail validation - it's deprecated.

This was done on https://github.com/Podcastindex-org/podcast-namespace/commit/9b1c3fb94d576f7b4d264791f465072729c9933b

And a new `podcast:image` tag was formalized with https://github.com/Podcastindex-org/podcast-namespace/pull/716

https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/tags/image.md - [permalink](https://github.com/Podcastindex-org/podcast-namespace/blob/336978220c14c0f180282ee0feb3e027567c4522/docs/tags/image.md)
